src/bot/commands/random/dad-jokes.ts

```diff
@@ -0,0 +1,60 @@
+import { Command, Input, Listener } from '@api';
+import * as request from 'request';
+import { Message } from 'discord.js';
+import { Emoji } from '@bot/libraries/emoji';
+import { Url } from 'url';
+import { monitorEventLoopDelay } from 'perf_hooks';
+import { CookieJar } from 'tough-cookie';
+const entities = require("html-entities").AllHtmlEntities;
+
+export class DadJokes extends Command {
+    constructor() {
+        super({
+            name: 'dadjoke',
+            description: 'Displays a random image of a cat.',
+            aliases: ["djoke", "djokes", "dadjokes"]
+        });
+    }
+
+    async execute(input: Input) {
+        let url = `https://icanhazdadjoke.com/`;
+
+        let message = await input.channel.send(`${Emoji.LOADING}  Fetching joke...`) as Message;
+        let headers = {
+            'Accept': 'application/json',
+            'User-Agent': 'Ember bot'
+        }
+
+        //Fetch from API
+        request({url, headers}, async (err, response, body) => {
+            // Handle HTTP errors
+            if (err) {
+                this.getLogger().error(err);
+                await message.edit(`${Emoji.ERROR}  Failed to get dad joke, try again later.`);
```

Comment:
This assumes that the message will still exist, which it might not. This has been the cause of a number of previous errors, so from now on please check to make sure the message exists before trying to edit it.